src/ingestion/utils.ts

```diff
@@ -163,3 +164,20 @@ export function timeoutGuard(message: string, timeout = defaultConfig.TASK_TIMEO
         Sentry.captureMessage(message)
     }, timeout)
 }
+
+const campaignParams = ['utm_source', 'utm_medium', 'utm_campaign', 'utm_content', 'utm_term']
+const initialParams = ['$browser', '$browser_version', '$current_url', '$os', '$referring_domain', '$referrer']
+
+/** If we get new UTM params, make sure we set those  **/
+export function personInitialAndUTMProperties(properties: Record<string, any>): Record<string, any> {
+    const propertiesCopy = { ...properties }
+    const maybeSet = Object.entries(properties).filter(([key, value]) => campaignParams.indexOf(key) > -1)
+    const maybeSetInitial = Object.entries(properties)
+        .filter(([key, value]) => [...campaignParams, ...initialParams].indexOf(key) > -1)
+        .map(([key, value]) => [`$initial_${key.replace('$', '')}`, value])
+    if (Object.keys(maybeSet).length > 0) {
+        propertiesCopy.$set = { ...(properties.$set || {}), ...Object.fromEntries(maybeSet) }
+        propertiesCopy.$set_once = { ...(properties.$set_once || {}), ...Object.fromEntries(maybeSetInitial) }
+    }
+    return propertiesCopy
+}
```

Comment:
Nit: Perhaps premature optimisation, yet since this will be run a lot of times, might be worth it.

I think it will be more efficient to store the `campaignParams` and `initialParams` as a `Set` and then just check `campaignParams.has(key)`. That should be an instant operation instead of looping through a bunch of arrays with `indexOf`.